### PR TITLE
feat: PriceList context and new PriceList UI

### PR DESCRIPTION
**Pull request by kasperkristensen**
**What**
- Adds PriceListContextProvider
- Adds form sections
- Adds create new PriceList UI
  - _Note_: I've not added the section for adding prices to the list, as this will be part of a seperate PR.

**Note**
- I assume that we are going to need to modify the `mappers` a bit when we implement functionality to Add/Edit prices, as from the UI we are going to need to group variants by `product_id`, which should not be sent to the backend, as it's not part of the request.

### CODE REVIEW

File: src/domain/pricing/pricing-form/sections/configuration.tsx [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+  }
+  if (config.ends_at) {
+    enabledConfigs.push(ConfigurationField.ENDS_AT)
+  }
+
+  return enabledConfigs
+}
+
+const Configuration: React.FC<ConfigurationProps> = () => {
+  const { customer_groups, isLoading } = useAdminCustomerGroups()
```
Comment (zakariaelas) [inline src/domain/pricing/pricing-form/sections/configuration.tsx:37]: thought (non-blocking): we probably need to implement some mechanism to handle pagination. Current core implementation only returns 10 customer groups, so we might run into issues sooner than elsewhere (i.e: regions). 

Found this [project](https://www.npmjs.com/package/react-select-async-paginate) that might help us implement this with `react-select`. I personally prefer we track this in a separate ticket and instead focus on getting this PR in 😄 . What do you think?

**Review by zakariaelas — APPROVED** [review]
LGTM! Great work @kasperkristensen 👍

I think it might be better to change the base branch to something other than `develop` as there are still some parts that will come from other PRs. I've used the `feat/price-list-details` branch as base in my PR. Might be worth considering it as I'll be reusing most of your work in the details page. What do you think?

File: src/domain/pricing/pricing-form/sections/configuration.tsx (hunk 1 above)
Comment (kasperkristensen) [reply src/domain/pricing/pricing-form/sections/configuration.tsx:37]: I was also thinking that we need to handle this somehow, the project you linked looks like a great candidate! But I agree we should do that in a separate PR 😄

**Comment by kasperkristensen** [comment]
@zakariaelas Are you fine with me merging it into `feat/price-list-details` or would you prefer something else? 😄

**Review by fPolic — APPROVED** [review]
LGTM nice work 👍

**Comment by zakariaelas** [comment]
@kasperkristensen `feat/price-list-details` is perfect!

File: src/domain/pricing/pricing-form/form-header/index.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+const FormHeader = (props: PriceListFormProps) => {
+  const { handleSubmit } = usePriceListForm()
+  const notification = useNotification()
+
+  const closeForm = () => {
+    navigate(-1)
+  }
+
+  const createPriceList = useAdminCreatePriceList()
+  const updatePriceList = useAdminUpdatePriceList(props?.id!)
```
[the remark this answers is not in the document]
Comment (kasperkristensen) [reply src/domain/pricing/pricing-form/form-header/index.tsx]: True! Removed the `?`, still need to use a bang notation to tell TS that the `id` always exists, in the cases where `updatePriceList` is used 😄

**Pull request merged**